### PR TITLE
Changed button label to button text, added prop and fixed examples

### PR DESCRIPTION
**Pull request by micahgodbolt**
Turns out that some people had both label AND child strings for their buttons, so we got this

image

Seeing as Label is actually an HTML attribute for <track> tags, decided to use an explicit `text` prop to be consistent with other components like dropdown. I also changed the logic so that if there is text, AND a string in the child, that only the text is used. Passing a component through as a child still works as expected.

imgo

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts [hunk 1]
```diff
@@ -39,6 +39,11 @@ export interface IButtonProps extends React.HTMLProps<HTMLButtonElement | HTMLAn
   ariaDescription?: string;
 
   /**
+  * Text to render button label. If text is supplied, it will override any string in button children. Other children components will be passed through after the text.
+  */
+  text?: string;
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: So the `label` prop that was in use before was from HTMLButtonElement, and not a prop we documented?

File: apps/fabric-examples/src/pages/ButtonPage/examples/Button.Primary.Example.tsx [hunk 2]
```diff
@@ -16,7 +16,8 @@ export class ButtonPrimaryExample extends React.Component<IButtonProps, {}> {
         <PrimaryButton
           data-automation-id='test'
           disabled={ disabled }
-        >Create account</PrimaryButton>
+          text="Create account"
+        >Text Overrides a Child String</PrimaryButton>
```
Comment (mikewheaton) [inline apps/fabric-examples/src/pages/ButtonPage/examples/Button.Primary.Example.tsx]: This is a good test case to have, but do we want it to be in the example? I would think in general people should be using the `text` prop **or** children and not both.

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts (hunk 1 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: correct. we'd thought that 'label' was a typical button tag for some reason. Still not sure why other people were using it in their product.

File: apps/fabric-examples/src/pages/ButtonPage/examples/Button.Primary.Example.tsx (hunk 2 above)
Comment (micahgodbolt) [inline apps/fabric-examples/src/pages/ButtonPage/examples/Button.Primary.Example.tsx]: Personally, I think it's handy to see and quite self explanatory if someone was to copy/paste it. I'm fine either way.

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts (hunk 1 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: ohh shoot. Blasted. Bummer. OK. good catch.

File: apps/fabric-examples/src/pages/ButtonPage/examples/Button.Primary.Example.tsx (hunk 2 above)
Comment (dzearing) [inline apps/fabric-examples/src/pages/ButtonPage/examples/Button.Primary.Example.tsx]: I don't think it's a great example, though I can sympathize with having examples to prove scenarios... But not a good one to follow ;)

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts [hunk 3]
```diff
@@ -39,6 +39,11 @@ export interface IButtonProps extends React.HTMLProps<HTMLButtonElement | HTMLAn
...
+
+  /**
    * The button icon shown in command or hero type.
    * Deprecated at v1.2.3, to be removed at >= v2.0.0. Use IconButton instead.
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: This comment should be removed. Icons can be displayed on any button.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: Same with description

**Pull request merged**